install.sh

```diff
@@ -114,124 +193,152 @@ else
+      if [ $(grep -c "usbcore.usbfs_memory_mb=" /etc/default/grub) -eq 0 ]; then # Memory Limit has not already been set
+        sudo sed -i 's/GRUB_CMDLINE_LINUX_DEFAULT="[^"]*/& usbcore.usbfs_memory_mb=1000/' /etc/default/grub
+        echo "Increased the usbfs memory limits in the default grub configuration. Updating grub"
+        sudo update-grub
+      else
+        echo -e "\e[33mWarn: usbfs memory limit is already set in /etc/default/grub in the following line:\e[0m"
+        echo "$(grep "usbcore.usbfs_memory_mb" /etc/default/grub)"
+        echo -e "\e[33mNo changes made, verify that usbfs_memory_mb is set to a minimum of 1000 and then try rebooting the computer\e[0m"
+      fi
 
-echo -e "\e[94mConfiguring network service, if needed\e[0m"
-# Check if the service file exists
-if [ -e "/etc/systemd/system/network-online.target.wants/systemd-networkd-wait-online.service" ]; then
-    # Check if TimeoutStartSec is present in the service file
-    if grep -q "TimeoutStartSec=2sec" "/etc/systemd/system/network-online.target.wants/systemd-networkd-wait-online.service"; then
-        echo "TimeoutStartSec is already present in /etc/systemd/system/network-online.target.wants/systemd-networkd-wait-online.service"
     else
-        # Add TimeoutStartSec after RemainAfterExit=yes
-        sudo sed -i '/RemainAfterExit=yes/a \'"TimeoutStartSec=2sec"'' "/etc/systemd/system/network-online.target.wants/systemd-networkd-wait-online.service"
-        echo "TimeoutStartSec added to /etc/systemd/system/network-online.target.wants/systemd-networkd-wait-online.service"
+      echo -e "\e[33mWarn: /etc/default/grub configuration file not found, no changes made. usbfs_memory_mb must be set manually.\e[0m"
+      echo -e "\e[33mSee https://github.com/ros-drivers/flir_camera_driver/blob/humble-devel/spinnaker_camera_driver/docs/linux_setup_flir.md for instructions\e[0m"
+      exit 0
     fi
+  else
+    echo "usbfs_memory_mb is already set to $val, no changes necessary."
+  fi
+
+  echo -e "\e[32mDone: Setting up groups\e[0m"
+  echo ""
+  echo -e "\e[32mClearpath Computer Installer Complete\e[0m"
+  echo -e "\e[94mTo continue installation visit: https://docs.clearpathrobotics.com/docs/ros/networking/computer_setup \e[0m"
 else
-    echo "Service file /etc/systemd/system/network-online.target.wants/systemd-networkd-wait-online.service not found."
+  echo -e "\e[32mClearpath Computer Installer needs to be ran as a user, please re-run.\e[0m"
 fi
 
-echo -e "\e[32mDone: Configuring network service\e[0m"
-echo ""
 
-echo -e "\e[32mClearpath Computer Installer Complete\e[0m"
-echo -e "\e[94mTo continue installation visit: https://docs.clearpathrobotics.com/docs/ros/networking/computer_setup \e[0m"
-echo -e "\e[94mTo start the robot service run:\e[0m sudo systemctl daemon-reload && sudo systemctl start clearpath-robot.service"
+# Reenable messages about restarting services in systems with needrestart installed
+if [ -d /etc/needrestart/conf.d ]; then
```

Comment:
Can we check if the file itself exists, not just the folder?

Reply:
Added.